src/main/java/uk/gov/homeoffice/digital/sas/timecard/repositories/TimeEntryRepository.java

```diff
@@ -13,18 +13,18 @@
 public interface TimeEntryRepository extends JpaRepository<TimeEntry, UUID> {
 
   @Query("SELECT t FROM time_entry t "
-      + "WHERE t.ownerId = cast(:ownerId as org.hibernate.type.UUIDCharType) "
-      + "AND (:id IS NULL OR t.id <> cast(:id as org.hibernate.type.UUIDCharType)) "
-      + "AND (:tenantId IS NULL OR t.tenantId = cast(:tenantId as org.hibernate.type.UUIDCharType))"
+      + "WHERE t.ownerId = :ownerId "
+      + "AND (cast(:id as string) IS NULL OR t.id <> cast(:id as string)) "
```

Comment:
Have you tested the second cast in this line? I am suspecting because `id` of type UUID the cast as string might fail

Reply:
If the second cast isn't there, you get this error: `ERROR: operator does not exist: character varying <> uuid`, which I think means it can't compare between a uuid and a character varying type?

Reply:
You can test it out on the branch deploy here: https://eahw-0000-timecard.dev.callisto-notprod.homeoffice.gov.uk/swagger-ui/index.html#/